lib/hanami/action/view_name_inferrer.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Hanami
+  class Action
+    class ViewNameInferrer
+      ALTERNATIVE_NAMES = {
+        "create" => "new",
+        "update" => "edit"
+      }.freeze
+
+      class << self
+        def call(action_name:, provider:)
+          application = provider.respond_to?(:application) ? provider.application : Hanami.application
```

Comment:
@timriley why is this needed?

Reply:
@solnic Because it's possible for the `provider` to be the `Hanami.application` itself, not just a slice. And at the moment `Hanami::Application` does not offer an `.application` method. So we need this to support actions that are in both the applications's root `/lib/` as well as within slices.

I have a card to create an `Hanami::Application.application` method (simply returning `self`) so we no longer need to do these `provider.respond_to?` checks: https://trello.com/c/Y7JWzo7y